File: Java/ptice.java
import java.io.*;
import java.util.*;

public class ptice {
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        String[] keys = { "ABC", "BABC", "CCAABB" };
        int n = sc.nextInt();
        String ans = sc.next();
        int[] count = new int[3];
        for (int x = 0; x < n; x++) {
            char c = ans.charAt(x);
            for (int i = 0; i < 3; i++)
                if (c == keys[i].charAt(x % keys[i].length()))
                    count[i]++;
        }
        int max = 0;
        for (int x = 0; x < 3; x++)
            max = Math.max(max, count[x]);
        System.out.println(max);
        for (int x = 0; x < 3; x++) {
            if (max == count[x]) {
                if (x == 0)
                    System.out.println("Adrian");
                else if (x == 1)
                    System.out.println("Bruno");
                else
                    System.out.println("Goran");
            }
        }
    }
}

File: C#/ptice.cs
using System;

public class ptice {
    public static void Main (string[] args) {
        string a = "ABC";
        string b = "BABC";
        string g = "CCAABB";
        int ac = 0, bc = 0, gc = 0;
        int i1 = 0, i2 = 0, i3 = 0;
        int high = 0;
        Console.ReadLine ();
        foreach (char c in Console.ReadLine ().ToCharArray ()) {
            if (c == a[i1])
                ac++;
            if (c == b[i2])
                bc++;
            if (c == g[i3])
                gc++;
            i1 = i1 == 2 ? 0 : i1 + 1;
            i2 = i2 == 3 ? 0 : i2 + 1;
            i3 = i3 == 5 ? 0 : i3 + 1;
            high = ac > high ? ac : bc > high ? bc : gc > high ? gc : high;
        }
        Console.WriteLine (high);
        if (ac == high)
            Console.WriteLine ("Adrian");
        if (bc == high)
            Console.WriteLine ("Bruno");
        if (gc == high)
            Console.WriteLine ("Goran");
    }
}